### PR TITLE
Significant performance improvement due to removal of deep clone operations inside of JSON diff/patch

### PR DESCRIPTION
**Pull request by clbond**
In this PR I have created a copy of a library that we depended upon, https://github.com/Starcounter-Jack/JSON-Patch, made a fork of it, made significant performance improvements to it, and then replaced our dependency on that library with our local fork.

The biggest performance improvement comes from the removal of the deep cloning operations that were happening inside the JSON patch library. These were completely unnecessary in the context of our use-case and introduced a significant performance bottleneck.

### CODE REVIEW

File: src/utils/patch.ts [hunk 1]
```diff
@@ -0,0 +1,284 @@
+// This code contains work that is copyrighted to Joachim Wester (2013)
```
Comment (igor-ka) [inline src/utils/patch.ts]: @clbond I am assuming that only the minimal code that is needed for Augury was forked into this file from the original repo?

Comment (clbond) [inline src/utils/patch.ts]: Yup, the bare minimum

Comment (igor-ka) [inline src/utils/patch.ts]: 👍

**Comment by igor-ka** [comment]
@clbond LGTM :shipit: 

PS. consider submitting a PR to the original repo if it makes sense

**Pull request merged**